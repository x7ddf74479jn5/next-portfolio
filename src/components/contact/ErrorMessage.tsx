import styles from "src/styles/components/contact/ContactForm.module.scss";

type Props = {
  children?: React.ReactNode;
  message?: string;
};

const ErrorMessage: React.VFC<Props> = ({ children, message }) => {
  return <p className={styles.error}>{children || message}</p>;
};

export default ErrorMessage;
