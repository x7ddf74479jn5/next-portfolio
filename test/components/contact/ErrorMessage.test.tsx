import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import ErrorMessage from "src/components/contact/ErrorMessage";

describe("ErrorMessage", () => {
  let renderResult: RenderResult;
  beforeEach(() => {
    renderResult = render(<ErrorMessage>test</ErrorMessage>);
  });
  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.container).toHaveTextContent("test");
  });
});
