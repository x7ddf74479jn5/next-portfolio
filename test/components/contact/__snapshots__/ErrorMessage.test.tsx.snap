// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="error"
  >
    test
  </p>
</DocumentFragment>
`;
